base/Plain_netCDF_Time.F90

```diff
@@ -511,3 +514,239 @@ subroutine convert_twostring_2_esmfinterval(symd, shms, interval, rc)
   end subroutine convert_twostring_2_esmfinterval
 
 end module Plain_NetCDF_Time
+
+
+module Fortran_read_file
+
+!  procedure :: matchbgn
+!  procedure :: matches
+!  procedure :: scan_begin
+!  procedure :: scan_contain
+!  generic :: scan_count_matchbgn
+!  generic :: go_last_pattern
+
+contains
+  subroutine scan_begin (iunps, substring, rew)
+    implicit none
+    ! unit of input
+    integer, intent(in) :: iunps
+    ! Label to be matched
+    character (len=*), intent(in) :: substring
+    logical, intent(in) :: rew
+    ! String read from file
+    character (len=100) :: line
+    ! Flag if .true. rewind the file
+    !logical, external :: matchbgn
+!    logical :: matchbgn
+    integer :: ios
+    !
+    ios = 0
+    if (rew) rewind (iunps)
+    do while (ios==0)
+       read (iunps, '(a100)', iostat = ios, err = 300) line
+       !       if (matchbgn (line, substring) ) return
+       !!write(6,*) 'line ', trim(line)
+       !!write(6,*) 'substring ', trim(substring)
+       if (matchbgn (trim(line), trim(substring)) ) return
+    enddo
+    return
+300 call error_nonstop ('scan_begin', &
+         'No '//trim(substring)//' block', abs (ios) )
+  end subroutine scan_begin
```

Comment:
Don't make the change, but you can instead have `read()` return an error and then use a conditional on the error rather than branching.  

Is the error trapping handled correctly?  Cannot tell as this does not use our `_RC` mechanism.  Very important that all I/O does robust error trapping.

Reply:
Right, I agree the _RC mechanism is better.   There is not an error trapping here, I was trying to just to get it work.

Reply:
ok - will approve for now and trust you'll circle back and fix that bit up later.

Reply:
Actually the call name is mis-leading, it is not an error when reading the file reaches the end line.   I donot track the erorr because it ACTUALLY is not an error, but a note that the file reaches the end.

Reply:
In that case, there are better options.   IOSTAT can be used instead of err, and you can check IOSTAT for EOF.   And stop/return if EOF but raise exception if somethinng else.